File: 3555-final-array-state-after-k-multiplication-operations-i/3555-final-array-state-after-k-multiplication-operations-i.java
class Solution {
    public int [] getFinalState (int [] nums, int k, int multiplier) {
        for (int i = 0; i < k; i++) {
            int index = getIndex (nums);
            nums [index] *= multiplier;
        }
        return nums;
    }

    private int getIndex (int [] array) {
        int min = Integer.MAX_VALUE;
        int index = 0;
        for (int i = 0; i < array.length; i++) {
            if (array [i] < min) {
                min = array [i];
                index = i;
            }
        }
        return index;
    }
}

File: 3555-final-array-state-after-k-multiplication-operations-i/3555-final-array-state-after-k-multiplication-operations-i.py
class Solution:
    def getFinalState(self, nums: List[int], k: int, multiplier: int) -> List[int]:
        for _ in range (k):
            i = self.get_index (nums)
            nums [i] *= multiplier
        return nums

    def get_index (self, array):
        min_val = float ('inf')
        index = 0
        for i in range (len (array)):
            if array [i] < min_val:
                min_val = array [i]
                index = i
        return index